File: fullstack-cms-system-plus/backend/src/main/java/com/cms/example/content/Content.java
```java
package com.cms.example.content;

import com.cms.example.category.Category;
import com.cms.example.user.User;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "contents")
public class Content {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(unique = true)
    private String slug; // URL-friendly version of the title

    @Lob // For larger text content
    @Column(nullable = false)
    private String body;

    private String featuredImage; // URL or path to featured image

    @Enumerated(EnumType.STRING)
    private ContentStatus status; // DRAFT, PUBLISHED, ARCHIVED

    private ContentType type; // POST, PAGE

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id", nullable = false)
    private User author;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    private LocalDateTime publishedAt;

    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        if (this.status == null) {
            this.status = ContentStatus.DRAFT;
        }
        if (this.type == null) {
            this.type = ContentType.POST;
        }
        if (this.slug == null || this.slug.isEmpty()) {
            this.slug = title.toLowerCase().replace(" ", "-").replaceAll("[^a-z0-9-]", "");
        }
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
        if (this.status == ContentStatus.PUBLISHED && this.publishedAt == null) {
            this.publishedAt = LocalDateTime.now();
        }
        if (this.slug == null || this.slug.isEmpty()) {
            this.slug = title.toLowerCase().replace(" ", "-").replaceAll("[^a-z0-9-]", "");
        }
    }
}
```

File: fullstack-cms-system-advanced/backend/src/entities/Content.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn, ManyToOne, ManyToMany, JoinTable } from 'typeorm';
import { User } from './User';
import { Category } from './Category';
import { Tag } from './Tag';

export enum ContentStatus {
  DRAFT = 'draft',
  PUBLISHED = 'published',
  ARCHIVED = 'archived',
}

@Entity('content')
export class Content {
  @PrimaryGeneratedColumn('uuid')
  id!: string;

  @Column({ nullable: false })
  title!: string;

  @Column({ unique: true, nullable: false })
  slug!: string;

  @Column('text', { nullable: false })
  body!: string;

  @Column({
    type: 'enum',
    enum: ContentStatus,
    default: ContentStatus.DRAFT,
  })
  status!: ContentStatus;

  @Column({ type: 'timestamp', nullable: true })
  publishedAt?: Date;

  @ManyToOne(() => User, (user) => user.content, { nullable: false, onDelete: 'CASCADE' })
  author!: User;

  @ManyToOne(() => Category, (category) => category.content, { nullable: true, onDelete: 'SET NULL', eager: true })
  category?: Category;

  @ManyToMany(() => Tag, (tag) => tag.content, { eager: true })
  @JoinTable({ name: 'content_tags' })
  tags!: Tag[];

  @CreateDateColumn()
  createdAt!: Date;

  @UpdateDateColumn()
  updatedAt!: Date;
}